File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/pkg/PackageClient.scala
// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.ledger.client.services.pkg

import com.daml.ledger.api.domain.LedgerId
import com.daml.ledger.api.v1.package_service._
import com.daml.ledger.api.v1.package_service.PackageServiceGrpc.PackageServiceStub

import scala.concurrent.Future

class PackageClient(val ledgerId: LedgerId, service: PackageServiceStub) {
  private[daml] val it = new withoutledgerid.PackageClient(service)

  def listPackages(token: Option[String] = None): Future[ListPackagesResponse] =
    it.listPackages(ledgerId, token)

  def getPackage(packageId: String, token: Option[String] = None): Future[GetPackageResponse] =
    it.getPackage(packageId, ledgerId, token)

  def getPackageStatus(
      packageId: String,
      token: Option[String] = None,
  ): Future[GetPackageStatusResponse] = it.getPackageStatus(packageId, ledgerId, token)
}

File: language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/PackageClient.java
// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.ledger.rxjava;

import com.daml.ledger.javaapi.data.GetPackageResponse;
import com.daml.ledger.javaapi.data.GetPackageStatusResponse;
import io.reactivex.Flowable;
import io.reactivex.Single;

/** An RxJava version of {@link com.daml.ledger.api.v1.PackageServiceGrpc} */
public interface PackageClient {
  Flowable<String> listPackages();

  Flowable<String> listPackages(String accessToken);

  Single<GetPackageResponse> getPackage(String packageId);

  Single<GetPackageResponse> getPackage(String packageId, String accessToken);

  Single<GetPackageStatusResponse> getPackageStatus(String packageId);

  Single<GetPackageStatusResponse> getPackageStatus(String packageId, String accessToken);
}
